tokio-console/src/state/mod.rs

```diff
@@ -538,3 +538,64 @@ fn pb_duration(dur: prost_types::Duration) -> Duration {
     let nanos = u64::try_from(dur.nanos).expect("duration should not be negative!");
     Duration::from_secs(secs) + Duration::from_nanos(nanos)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    #[cfg(target_os = "linux")]
+    fn test_format_location() {
+        let mut location1 = proto::Location::default();
+        location1.file = Some(
+            "/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.0.1/src/lib.rs"
+                .to_string(),
+        );
+        let mut location2 = proto::Location::default();
+        location2.file = Some("/home/user/.cargo/git/checkouts/tokio-1.0.1/src/lib.rs".to_string());
+        let mut location3 = proto::Location::default();
+        location3.file = Some("/home/user/projects/tokio-1.0.1/src/lib.rs".to_string());
+
+        assert_eq!(
+            format_location(Some(location1)),
+            "<cargo>/tokio-1.0.1/src/lib.rs"
+        );
+        assert_eq!(
+            format_location(Some(location2)),
+            "<cargo>/tokio-1.0.1/src/lib.rs"
+        );
+        assert_eq!(
+            format_location(Some(location3)),
+            "/home/user/projects/tokio-1.0.1/src/lib.rs"
+        );
+        assert_eq!(format_location(None), "<unknown location>");
+    }
+
+    #[test]
+    #[cfg(target_os = "macos")]
```

Comment:
is it actually necessary to only compile these tests on the OS whose paths they test? since we don't use `std::path::Path` to construct or parse the paths, i think the behavior should be the same on all operating systems — we're just testing how the console handles strings that _look_ like the paths that are produced by those operating systems. i'd prefer it if all the tests were always run, regardless of what platform a developer runs them on. 

also, it's worth noting that these paths are generated when the instrumented application is compiled, not from the system that's running the console. so, for example, if a console running on a Windows dev machine connects to an application compiled on Linux, it will see Linux-style paths, not Windows ones. therefore, i think we ought to be testing against all paths, regardless of the OS we're compiling the console for.

Reply:
Yeah. Sounds make sense.